File: 02Algorithm_Study/03_week_homeWork/Greedy_Big_Number.cpp
#include <string>
#include <vector>
 
using namespace std;
 
//string solution(string number, int k) {
//    string answer = "";
// 
//    int numSize = number.size() - k;
//    int start = 0;
//    for(int i=0; i<numSize; i++) {
//        char maxNum = number[start];
//        int maxIdx = start;
//        cout << "start : "<< start << '\n';
//        for(int j=start; j<=k+i; j++) {
//            if(maxNum < number[j]) {
//                maxNum = number[j];
//                maxIdx = j;
//            }
//        }
//        start = maxIdx + 1;
//        answer += maxNum;
//    }
// 
//    return answer;
//}

#include <string>
#include <vector>
 
using namespace std;
string solution(string number, int k){
	string answer = "";
	
	int size = number.size() - k;
	int start = 0;
	for(int i = 0 ; i< size; i ++){
		char CharNum = number[start];
		int NumMax = start;
		for(int j = start; j <= k + i ; j ++){
			if(CharNum<number[j]){
				CharNum = number[j];
				NumMax = j;
			}
		}
			start = NumMax + 1;
			answer += CharNum;
	}
	return answer;
}

File: 02Algorithm_Study/03_week_homeWork/Greedy_Big_Number.py
def solution(number,k):

    answer = [number[0]]

    for num in number[1:]:
        while len(answer)>0 and answer[-1]<num and k>0:
            k-=1
            answer.pop()
        answer.append(num)
    if(k!=0):
        answer = answer[:-k]


    answer = ''.join(answer)
        
    
    return answer

#https://jjluveeecom.tistory.com/10
print(solution("4177252841",4))
        
                
                
